src/services/products.js

```diff
@@ -0,0 +1,102 @@
+const { Product } = require('../models');
+const { productImageService } = require('../uploads');
+
+class ProductsService {
+  async createProduct (req, res) {
+    await productImageService.uploadImage(req);
+    const product = await new Product(req.body).save();
+
+    return res.json(product);
+  }
+
+  async updateProduct (req, res) {
+    const available = await Product.findById(req.params.id).exec();
+
+    if (req.files && available) {
+      if (available.image.length !== 0) {
+        await productImageService.deleteImage(available.image);
+      }
+
+      await productImageService.uploadImage(req);
+    }
+
+    const product = await Product.findByIdAndUpdate(
+      req.params.id,
+      req.body
+    ).exec();
+
+    return product ? res.json(true) : res.boom.notFound();
+  }
+
+  async getProducts (req, res) {
+    const { _id, name, category, sizes, status, minPrice, maxPrice } =
+            (typeof req.query !== 'undefined' && req.query) || {};
+    const query = {};
+
+    if (_id != null) {
+      query._id = _id;
+    }
+    if (name != null) {
```

Comment:
What a big construction... I meant to use something like that: https://kb.objectrocket.com/mongo-db/mongoose-partial-text-search-606

Reply:
Fixed